pkg/controller/unifiedpushserver/unifiedpushserver_controller.go

```diff
@@ -560,77 +561,90 @@ func (r *ReconcileUnifiedPushServer) Reconcile(request reconcile.Request) (recon
 				}
 			}
-		}
 
-		desiredImage := constants.PostgresImage
+			desiredImage := constants.PostgresImage
 
-		containerSpec := findContainerSpec(foundPostgresqlDeployment, cfg.PostgresContainerName)
-		if containerSpec == nil {
-			reqLogger.Info("Unable to do image reconcile: Unable to find container spec in deployment", "Deployment.Namespace", foundPostgresqlDeployment.Namespace, "Deployment.Name", foundPostgresqlDeployment.Name, "ContainerSpec", cfg.PostgresContainerName)
-			return reconcile.Result{Requeue: true}, nil
-		} else if containerSpec.Image != desiredImage {
-			reqLogger.Info("Container spec in deployment is using a different image. Going to update it now.", "Deployment.Namespace", foundPostgresqlDeployment.Namespace, "Deployment.Name", foundPostgresqlDeployment.Name, "ContainerSpec", cfg.PostgresContainerName, "ExistingImage", containerSpec.Image, "DesiredImage", desiredImage)
+			containerSpec := findContainerSpec(foundPostgresqlDeployment, cfg.PostgresContainerName)
+			if containerSpec == nil {
+				reqLogger.Info("Unable to do image reconcile: Unable to find container spec in deployment", "Deployment.Namespace", foundPostgresqlDeployment.Namespace, "Deployment.Name", foundPostgresqlDeployment.Name, "ContainerSpec", cfg.PostgresContainerName)
+				return reconcile.Result{Requeue: true}, nil
+			} else if containerSpec.Image != desiredImage {
+				reqLogger.Info("Container spec in deployment is using a different image. Going to update it now.", "Deployment.Namespace", foundPostgresqlDeployment.Namespace, "Deployment.Name", foundPostgresqlDeployment.Name, "ContainerSpec", cfg.PostgresContainerName, "ExistingImage", containerSpec.Image, "DesiredImage", desiredImage)
 
-			// update
-			updateContainerSpecImage(foundPostgresqlDeployment, cfg.PostgresContainerName, desiredImage)
+				// update
+				updateContainerSpecImage(foundPostgresqlDeployment, cfg.PostgresContainerName, desiredImage)
 
-			// enqueue
-			err = r.client.Update(context.TODO(), foundPostgresqlDeployment)
-			if err != nil {
-				reqLogger.Error(err, "Failed to update Deployment", "Deployment.Namespace", foundPostgresqlDeployment.Namespace, "Deployment.Name", foundPostgresqlDeployment.Name)
-				return reconcile.Result{}, err
+				// enqueue
+				err = r.client.Update(context.TODO(), foundPostgresqlDeployment)
+				if err != nil {
+					reqLogger.Error(err, "Failed to update Deployment", "Deployment.Namespace", foundPostgresqlDeployment.Namespace, "Deployment.Name", foundPostgresqlDeployment.Name)
+					return reconcile.Result{}, err
+				}
+				return reconcile.Result{Requeue: true}, nil
 			}
-			return reconcile.Result{Requeue: true}, nil
 		}
 	}
 	//#endregion
 
 	//#region Postgres Service
-	postgresqlService, err := newPostgresqlService(instance)
-	if err != nil {
-		return reconcile.Result{}, err
+	var postgresqlService *corev1.Service
+	if externaldb {
+		postgresqlService, err = newExternalPostgresqlService(instance)
```

Comment:
Even though we're creating this Service in the case where externaldb is true, I don't believe it's actually being used. Remember that UPS consumes the host from an environment variable that's populated from the secret, and the Secret contains the actual external host in this case, and not this service.

Given that, I'd say we don't need to create this Service when externaldb is true. WDYT?

Reply:
sounds good, will make change, test it and rebase